SSZipArchive/SSZipArchive.m

```diff
@@ -965,38 +963,8 @@ - (BOOL)close
 #pragma mark - Private
 
 + (NSString *)_filenameStringWithCString:(const char *)filename
-                         version_made_by:(uint16_t)version_made_by
-                    general_purpose_flag:(uint16_t)flag
                                     size:(uint16_t)size_filename {
     
-    // Respect Language encoding flag only reading filename as UTF-8 when this is set
-    // when file entry created on dos system.
-    //
-    // https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
-    //   Bit 11: Language encoding flag (EFS).  If this bit is set,
-    //           the filename and comment fields for this file
-    //           MUST be encoded using UTF-8. (see APPENDIX D)
-    uint16_t made_by = version_made_by >> 8;
-    BOOL made_on_dos = made_by == 0;
-    BOOL languageEncoding = (flag & (1 << 11)) != 0;
-    if (!languageEncoding && made_on_dos) {
-        // APPNOTE.TXT D.1:
-        //   D.2 If general purpose bit 11 is unset, the file name and comment should conform
-        //   to the original ZIP character encoding.  If general purpose bit 11 is set, the
-        //   filename and comment must support The Unicode Standard, Version 4.1.0 or
-        //   greater using the character encoding form defined by the UTF-8 storage
-        //   specification.  The Unicode Standard is published by the The Unicode
-        //   Consortium (www.unicode.org).  UTF-8 encoded data stored within ZIP files
-        //   is expected to not include a byte order mark (BOM).
-        
-        //  Code Page 437 corresponds to kCFStringEncodingDOSLatinUS
-        NSStringEncoding encoding = CFStringConvertEncodingToNSStringEncoding(kCFStringEncodingDOSLatinUS);
-        NSString* strPath = [NSString stringWithCString:filename encoding:encoding];
-        if (strPath) {
-            return strPath;
-        }
-    }
-    
```

Comment:
The purpose of this code block is to guanartee that the files can be unzipped correctly, the files that are encode with non-utf8 format and zipped in windows platform.

But saddly, most zip tool in windows never set the utf8 flag even the file name is encoded with utf8 format. And that is the reason why the file names become messy code after processed by this code block.

Reply:
I guess I'll have to get my hands on a Windows machine for verification.